### PR TITLE
Tool Preview

### LINKED ISSUE #55: Vorschau für Tools (8)
Als NutzerIn möchte ich eine visuelle Vorschau für die aktuell mögliche Bearbeitung sehen, um einschätzen zu können, wie sich ein Klick auf die Zeichnung auswirkt. Insbesondere sollen dabei verschiedene Pinselstärken unterschiedlich dargestellt werden, z. B. durch eine Kreisdarstellung in Größe der Stärke.

### LINKED ISSUE #84: Paint bucket cursor not transparent
The paint bucket cursor should have a transparent background.

### PR DESCRIPTION
**Pull request by DevSchmidtchen**
Closes #55
Closes #84

### CODE REVIEW

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/chooseFont.st [hunk 1]
```diff
@@ -1,3 +1,4 @@
 tool selection
 chooseFont
+	self currentCursor: nil.
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/chooseFont.st]: This probably leads to the problem that, after choosing the font, no context cursor is shown.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/clearCanvas.st [hunk 2]
```diff
@@ -1,4 +1,6 @@
 tool selection
 clearCanvas
-	self switchMode.
+	self
+		switchMode;
+		currentCursor: nil.
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/clearCanvas.st]: This probably leads to no context cursor being shown after clearing the canvas.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/drawEraser.st [hunk 3]
```diff
@@ -5,4 +5,5 @@ drawEraser
 		isErasing: true;
 		mode: #pen;
 		currentTool: #drawEraser;
+		currentCursor: (Form dotOfSize: self eraserSize color: Color white);
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/drawEraser.st]: I don't really like the concept of a white cursor on a mostly white background. When there is not much color on the canvas, one can barely find the cursor.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/drawTriangle.st [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 drawTriangle
-	self pen sourceForm: self prepareTrianglePenSource.
+	self pen sourceForm: (self prepareTrianglePenSource offset: (self penNibOffset: self brushSize asPoint)).
 	self
 		isErasing: false;
 		mode: #pen;
 		currentTool: #drawTriangle;
-		switchMode
+		currentCursor: ((Form extent: self brushSize asPoint depth: self defaultDepth) fillShape: self prepareTrianglePenSource fillColor: self currentColor);
+		switchMode
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/drawTriangle.st]: Missing newline.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/fillShape.st [hunk 5]
```diff
@@ -4,4 +4,5 @@ fillShape
 		isErasing: false;
 		mode: #fill;
 		currentTool: #fillShape;
+		currentCursor: M2FillArea icon;
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/fillShape.st]: This cursor is not transparent.

File: packages/SketchMorph2-Core.package/M2Canvas.class/instance/mouseLeave..st [hunk 6]
```diff
@@ -0,0 +1,4 @@
+event handling
+mouseLeave: evt
+	evt hand showTemporaryCursor: nil
+
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2Canvas.class/instance/mouseLeave..st]: One line too much.

File: packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/cursor.st [hunk 7]
```diff
@@ -0,0 +1,3 @@
+accessing
+cursor
+	^ self subclassResponsibility
```
Comment (ClFeSc) [inline packages/SketchMorph2-Core.package/M2IconMenuItem.class/class/cursor.st]: This subclass responsibility seems to be not implemented.

**Review by ClFeSc — CHANGES_REQUESTED** [review]

**Review by ClFeSc — APPROVED** [review]

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/drawCircle.st [hunk 8]
```diff
@@ -5,4 +5,5 @@ drawCircle
 		isErasing: false;
 		mode: #pen;
 		currentTool: #drawCircle;
+		currentCursor: (Form dotOfSize: self brushSize color: self penColor);
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/drawCircle.st]: Maybe we should add a cursor for that like we did for `drawEraser` to make the code more consistent.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/drawRectangle.st [hunk 9]
```diff
@@ -7,4 +7,7 @@ drawRectangle
 		isErasing: false;
 		mode: #pen;
 		currentTool: #drawRectangle;
+		currentCursor: ((Form extent: self rectangleBrushSize depth: self defaultDepth)
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/drawRectangle.st]: Maybe we should add a cursor for that like we did for `drawEraser` to make the code more consistent.

File: packages/SketchMorph2-Core.package/M2Backend.class/instance/drawTriangle.st [hunk 10]
```diff
@@ -1,8 +1,9 @@
 tool selection
 drawTriangle
-	self pen sourceForm: self prepareTrianglePenSource.
+	self pen sourceForm: (self prepareTrianglePenSource offset: (self penNibOffset: self brushSize asPoint)).
 	self
 		isErasing: false;
 		mode: #pen;
 		currentTool: #drawTriangle;
+		currentCursor: ((Form extent: self brushSize asPoint depth: self defaultDepth) fillShape: self prepareTrianglePenSource fillColor: self currentColor);
```
Comment (maximilian-franz) [inline packages/SketchMorph2-Core.package/M2Backend.class/instance/drawTriangle.st]: Maybe we should add a cursor for that like we did for `drawEraser` to make the code more consistent.

**Review by kpostnov — APPROVED** [review]

**Review by maximilian-franz — APPROVED** [review]

**Review by jannis-hpi — APPROVED** [review]

**Review by johannes-poetzsch — APPROVED** [review]

**Pull request merged**